faucet/pages/api/mint.ts

```diff
@@ -0,0 +1,70 @@
+import { IncomingMessage, ServerResponse } from 'http'
+import { parse as parseUrl } from 'url'
+import { addresses } from '@hoprnet/hopr-ethereum'
+import networksConfig from '@hoprnet/hopr-ethereum/truffle-networks.json'
+import HoprFaucetAbi from '@hoprnet/hopr-ethereum/build/extracted/abis/HoprFaucet.json'
+import Web3 from 'web3'
+
+const { isAddress, toChecksumAddress, toWei } = Web3.utils
+const { PRIVATE_KEY, INFURA } = process.env
+
+export default async (req: IncomingMessage, res: ServerResponse) => {
+  try {
+    const query = parseUrl(req.url, true).query as {
+      network: addresses.Networks
+      address: string
+    }
+
+    // validate network
+    const network = query.network
+    if (typeof addresses.HOPR_FAUCET[network] === 'undefined') {
+      res.statusCode = 400
+      res.end('Invalid network.')
+      return
+    }
```

Comment:
Perhaps we could reduce the different scenarios that we are trying to get by and always return a `statusCode` and `message`? Similar to what `redux` does with the actions.